cpp/src/io/csv/writer_impl.cu

```diff
@@ -282,11 +282,17 @@ void writer::impl::write_chunked_begin(table_view const& table,
                                        const table_metadata* metadata,
                                        rmm::cuda_stream_view stream)
 {
-  if ((metadata != nullptr) && (options_.is_enabled_include_header())) {
+  if (options_.is_enabled_include_header()) {
     auto const& column_names = metadata->column_names;
     CUDF_EXPECTS(column_names.size() == static_cast<size_t>(table.num_columns()),
                  "Mismatch between number of column headers and table columns.");
 
+    // column_names should be filled out,generated, if metadata == nullptr
+    std::vector<std::string> col_names;
```

Comment:
`col_names `is never used. We can use it as a replacement for `metadata->column_names` as such:
`auto const& column_names = metadata == nullptr ? col_names : metadata_column_names`.